backend/src/main/java/hanglog/share/service/SharedTripService.java

```diff
@@ -0,0 +1,74 @@
+package hanglog.share.service;
+
+import static hanglog.global.exception.ExceptionCode.INVALID_SHARE_CODE;
+import static hanglog.global.exception.ExceptionCode.NOT_FOUND_SHARED_CODE;
+import static hanglog.global.exception.ExceptionCode.NOT_FOUND_TRIP_ID;
+import static hanglog.share.domain.type.SharedStatusType.UNSHARED;
+
+import hanglog.global.exception.BadRequestException;
+import hanglog.share.domain.SharedTrip;
+import hanglog.share.domain.repository.SharedTripRepository;
+import hanglog.share.domain.type.SharedStatusType;
+import hanglog.share.dto.request.SharedTripStatusRequest;
+import hanglog.share.dto.response.SharedTripCodeResponse;
+import hanglog.trip.domain.City;
+import hanglog.trip.domain.Trip;
+import hanglog.trip.domain.TripCity;
+import hanglog.trip.domain.repository.TripCityRepository;
+import hanglog.trip.domain.repository.TripRepository;
+import hanglog.trip.dto.response.TripDetailResponse;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class SharedTripService {
+
+    private final SharedTripRepository sharedTripRepository;
+    private final TripRepository tripRepository;
+    private final TripCityRepository tripCityRepository;
+
+    public TripDetailResponse getTripDetail(final String sharedCode) {
+        final SharedTrip sharedTrip = sharedTripRepository.findBySharedCode(sharedCode)
+                .orElseThrow(() -> new BadRequestException(NOT_FOUND_SHARED_CODE));
+
+        validateSharedStatus(sharedTrip.getSharedStatus());
+
+        final Long tripId = sharedTrip.getTrip().getId();
+        final Trip trip = tripRepository.findById(tripId)
+                .orElseThrow(() -> new BadRequestException(NOT_FOUND_TRIP_ID));
+        final List<City> cities = getCitiesByTripId(tripId);
+
+        return TripDetailResponse.of(trip, cities);
+    }
+
+    private void validateSharedStatus(final SharedStatusType sharedStatusType) {
+        if (sharedStatusType == UNSHARED) {
+            throw new BadRequestException(INVALID_SHARE_CODE);
+        }
+    }
+
```

Comment:
이번에도 validateSharedStatus 메서드 분리가 납득이 잘 가지 않아요. (개인적 의견입니다. 다른 사람들은 괜찮을 수도 있음)

1. 분리한 메서드의 로직이 복잡한가? ( = 긴가?)
   -> 짧고 간단합니다.
3. 분리한 메서드의 네이밍을 통해 해당 행위의 가독성을 증가시킬 수 있는가?
   -> SharedStatus를 검증한다는 네이밍 
   -> if문의 조건절 `(sharedStatusType == UNSHARED)` 이 이미 충분히 그 역할을 해주고 있지 않나.?

```suggestion
public TripDetailResponse getTripDetail(final String sharedCode) {
        final SharedTrip sharedTrip = sharedTripRepository.findBySharedCode(sharedCode)
                .orElseThrow(() -> new BadRequestException(NOT_FOUND_SHARED_CODE));
        
        if (sharedTrip.getSharedStatus() == UNSHARED) {
            throw new BadRequestException(INVALID_SHARE_CODE);
        }
        
        final Long tripId = sharedTrip.getTrip().getId();
        final Trip trip = tripRepository.findById(tripId)
                .orElseThrow(() -> new BadRequestException(NOT_FOUND_TRIP_ID));
        final List<City> cities = getCitiesByTripId(tripId);

        return TripDetailResponse.of(trip, cities);
    }
```

제가 봤을땐 해당 구문을 다시 안으로 넣어놔도 이해하는데 전혀 문제 없는데? 라는 생각이 들었습니다.

Reply:
검증 로직 같은 경우에는 명시적으로 보여주는것이 좋지 않은가 하여 분리를 하였습니다. 근데... 디노 말대로 어려운 로직도 아니긴 해서 분리해도 상관 없을것 같네요.